ovc5/scripts/install_sd.sh

```diff
@@ -0,0 +1,273 @@
+  2. Install (adjust for version changes):
+    ./petalinux-v2020.2-final-installer.run -d petalinux_sdk/ -p "arm aarch64"
+  3. Source:
+    source petalinux_sdk/settings.sh
+
+If installed, make sure petalinux/settings.sh is sourced.
+
+EOF
+    exit
+  fi
+
+  cp $XSA_PATH $PETALINUX_DIR/system.xsa
+  cd $PETALINUX_DIR
+  petalinux-config --get-hw-description --silentconfig
+  petalinux-build
+  petalinux-package \
+    --boot \
+    --force \
+    --fsbl $BOOT_FILES_DIR/zynq_fsbl.elf \
+    --fpga \
+    --u-boot
+}
+
+format_sd () {
+  sudo dd if=/dev/zero of=$target_device bs=1024 count=1
+
+  # Pulled straight from Stack Exchange with modifications for linux:
+  #     https://superuser.com/questions/332252/how-to-create-and-format-a-partition-using-a-bash-script
+  # to create the partitions programatically (rather than manually)
+  # we're going to simulate the manual input to fdisk
+  # The sed script strips off all the comments so that we can 
+  # document what we're doing in-line with the actual commands
+  # Note that a blank line (commented as "defualt" will send a empty
+  # line terminated with a newline to take the fdisk default.
+  sed -e 's/\s*\([\+0-9a-zA-Z]*\).*/\1/' << EOF | sudo fdisk ${target_device}
+    o # clear the in memory partition table
+    n # new partition
+    p # primary partition
+    1 # partition number 1
+      # default - start at beginning of disk 
+    +200M # 200 MB boot parttion
+    n # new partition
+    p # primary partition
+    2 # partion number 2
+      # default, start immediately after preceding partition
+      # default, extend partition to end of disk
+    a # make a partition bootable
+    1 # bootable partition is partition 1 -- /dev/sda1
+    t # change partition type
+    1 # partition 1
+    c # set to FAT32 (LBA)
+    t # change partition type
+    2 # partition 2
+    83 # magic linux partition number
+    p # print the in-memory partition table
+    w # write the partition table
+    q # and we're done
+EOF
+
+  sudo partprobe $target_device
+  
+  # First, a FAT32 filesystem named boot in the first partition:
+  sudo mkfs.vfat -F 32 -n boot ${target_device}1
+  # Next, an ext4 filesystem named root in the second partition:
+  sudo mkfs.ext4 -L root ${target_device}2
+}
+
+umount_drive () {
+  # Unmount boot partition.
+  sudo umount $BOOT_DIR
+  sudo rmdir $BOOT_DIR
+
+  # Unmount root partition.
+  sudo umount $ROOT_DIR
+  sudo rmdir $ROOT_DIR
+
+  # Remove mount directory.
+  sudo rmdir $MOUNT_DIR
+}
+
+mount_drive () {
+  # Mount boot partition.
+  sudo mkdir -p $BOOT_DIR
+  sudo mount ${target_device}1 $BOOT_DIR
+
+  # Mount root partition.
+  sudo mkdir -p $ROOT_DIR
+  sudo mount ${target_device}2 $ROOT_DIR
+}
+
+copy_bin () {
+  sudo cp $BOOT_FILES_DIR/BOOT.BIN $BOOT_DIR/boot.bin
+  sudo cp $BOOT_FILES_DIR/image.ub $BOOT_DIR
+  sudo cp $BOOT_FILES_DIR/boot.scr $BOOT_DIR
+}
+
+install_debian () {
+  sudo apt install qemu-user-static debootstrap debian-archive-keyring schroot
```

Comment:
No strong feelings here but should we have a -y to avoid this blocking or should we still get the users input?

Reply:
Since this is installing to the local system I'd rather the user get to see what's being installed haha.